src/mcp.rs

```diff
@@ -0,0 +1,186 @@
+    async fn litra_brightness_up(&self, Parameters(params): Parameters<LitraBrightnessParams>) -> Result<CallToolResult, McpError> {
+        match handle_brightness_up_command(params.serial_number.as_deref(), params.value, params.percentage) {
+            Ok(()) => Ok(CallToolResult::success(vec![Content::text("Brightness increased successfully")])),
+            Err(e) => Ok(CallToolResult::error(vec![Content::text(e.to_string())])),
+        }
+    }
+
+    #[tool(description = "Decrease the brightness of your Logitech Litra device")]
+    async fn litra_brightness_down(&self, Parameters(params): Parameters<LitraBrightnessParams>) -> Result<CallToolResult, McpError> {
+        match handle_brightness_down_command(params.serial_number.as_deref(), params.value, params.percentage) {
+            Ok(()) => Ok(CallToolResult::success(vec![Content::text("Brightness decreased successfully")])),
+            Err(e) => Ok(CallToolResult::error(vec![Content::text(e.to_string())])),
+        }
+    }
+
+    #[tool(description = "Set the temperature of your Logitech Litra device")]
+    async fn litra_temperature(&self, Parameters(params): Parameters<LitraTemperatureParams>) -> Result<CallToolResult, McpError> {
+        match handle_temperature_command(params.serial_number.as_deref(), params.value) {
+            Ok(()) => Ok(CallToolResult::success(vec![Content::text("Temperature set successfully")])),
+            Err(e) => Ok(CallToolResult::error(vec![Content::text(e.to_string())])),
+        }
+    }
+
+    #[tool(description = "Increase the temperature of your Logitech Litra device")]
+    async fn litra_temperature_up(&self, Parameters(params): Parameters<LitraTemperatureParams>) -> Result<CallToolResult, McpError> {
+        match handle_temperature_up_command(params.serial_number.as_deref(), params.value) {
+            Ok(()) => Ok(CallToolResult::success(vec![Content::text("Temperature increased successfully")])),
+            Err(e) => Ok(CallToolResult::error(vec![Content::text(e.to_string())])),
+        }
+    }
+
+    #[tool(description = "Decrease the temperature of your Logitech Litra device")]
+    async fn litra_temperature_down(&self, Parameters(params): Parameters<LitraTemperatureParams>) -> Result<CallToolResult, McpError> {
+        match handle_temperature_down_command(params.serial_number.as_deref(), params.value) {
+            Ok(()) => Ok(CallToolResult::success(vec![Content::text("Temperature decreased successfully")])),
+            Err(e) => Ok(CallToolResult::error(vec![Content::text(e.to_string())])),
+        }
+    }
+
+    #[tool(description = "List Logitech Litra devices connected to your computer")]
+    async fn litra_devices(&self) -> Result<CallToolResult, McpError> {
+        use litra::Litra;
+        
+        let context = Litra::new().map_err(|e| McpError::internal_error(e.to_string(), None))?;
+        let litra_devices: Vec<DeviceInfo> = context
```

Comment:
Surely it should be possible to reuse some CLI code for this? It already has the ability to output all devices asJSON for example